node/storage/hub/hub.py

```diff
@@ -246,6 +257,56 @@ async def list_memory_modules(self, memory_module_name=None) -> List:
     async def create_agent(self, agent_config: Dict) -> Tuple[bool, Optional[Dict]]:
         return await self.surrealdb.create("agent", agent_config)
 
+    def prepare_batch_query(self, secret_config: List[SecretInput], existing_data: List, update:bool = False) -> str:
+        existing_data_dict = {secret['key_name'] for secret in existing_data if 'key_name' in secret}
+        records_to_insert = []
+        records_to_update = []
+        
+        for secret in secret_config:
+            user_id = secret.user_id
+            key_name = secret.key_name
+            key_value = secret.secret_value
+
+            if not existing_data_dict or key_name not in existing_data_dict:
+                records_to_insert.append(f'{{"user_id": {user_id}, "key_name": "{key_name}", "secret_value": "{key_value}"}}')
+            else:
+                if update:
+                    records_to_update.append(
+                        {"secret_value": key_value, "user_id": user_id, "key_name": key_name}
+                    )
+
+        insert_query = ""
+        if records_to_insert:
+            insert_query = f"INSERT INTO api_secrets [{', '.join(records_to_insert)}];"
+
+        update_query = ""
+        if records_to_update:
+            for record in records_to_update:
+                update_query += f"UPDATE api_secrets SET secret_value='{record['secret_value']}' WHERE user_id={record['user_id']} AND key_name='{record['key_name']}';\n"
+
+        final_query = f"{insert_query}\n{update_query}".strip()
```

Comment:
is there an equivalent of prepared statements that we should be using? Generally using f-strings to build database queries with literal interpolation is a bad idea since it can create injection vectors.